src/schema.ts

```diff
@@ -0,0 +1,385 @@
+	bpm: number;
+	/** Whether or not the programming wheel is spinning. */
+	flywheelConnected: boolean;
+}
+
+/**
+ * All possible events relating to the machine itself.
+ */
+export type MachineEvent = MachineMuteEvent;
+/** An event representing the muting or unmuting of a [[Channel]] */
+export interface MachineMuteEvent {
+	kind: "machine_mute";
+	/** The channel affected. */
+	channel: Channel;
+	/** Whether or not the channel should be muted or unmuted. */
+	muted: boolean;
+}
+/**
+ * An event representing a change in the machine's BPM.
+ *
+ * This change will be instantaneous, so if a gradual change is
+ * desired, it must be achieved by specifying small increments of
+ * change to the BPM. This may change in the future.
+ */
+export interface MachineTempoEvent {
+	kind: "machine_tempo";
+	/** The BPM to set the machine to. */
+	bpm: number;
+}
+/**
+ * An event representing the flywheel getting connected or disconnected.
+ *
+ * This equates to the programming wheel starting or stopping instantaneously.
+ */
+export interface FlywheelConnectedEvent {
+	kind: "machine_flywheelConnected";
+	/**
+	 * Whether or not the flywheel is connected and thus also
+	 * whether or not the programming wheel is turning.
+	 */
+	connected: boolean;
+}
+
+/**
+ * Represents every channel that can be played on the vibraphone.
+ *
+ * We decided to start at one since the [[BassString]] type also
+ * starts at one. We also decided to only represent playable
+ * channels and not actual channels (two per note) because that is
+ * an implementation detail of the physical MMX.
+ */
+export type VibraphoneChannel = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+
+/**
+ * Represents the state pertaining to the vibraphone.
+ */
+export interface VibraphoneState {
+	/** Whether or not the vibrato is engaged */
+	vibratoEnabled: boolean;
+	/** The vibrato speed from 0 to 1 */
+	vibratoSpeed: number;
+	/**
+	 * Which notes represent which channels.
+	 * These cannot be changed via events and thus stay
+	 * the same throughout the course of an entire [[Performance]].
+	 * They can however, be changed while editing a [[Program]].
+	 */
+	notes: { [VC in VibraphoneChannel]: Note };
+}
+
+/**
+ * Any event pertaining to the vibraphone.
+ */
+export type VibraphoneEvent =
+	| VibraphoneVibratoEnabledEvent
+	| VibraphoneVibratoSpeedEvent;
+/** An event corresponding to a change in the [[VibraphoneState.vibratoEnabled]] property. */
+export interface VibraphoneVibratoEnabledEvent {
+	kind: "vibraphone_vibrato_enabled";
+	/** Whether or not the vibrato should be enabled. */
+	vibratoEnabled: boolean;
+}
+/** An event corresponding to a change in the [[VibraphoneState.vibratoSpeed]] property */
+export interface VibraphoneVibratoSpeedEvent {
+	kind: "vibraphone_vibrato_speed";
+	/** The new speed of the vibrato. */
+	vibratoSpeed: number;
+}
+
+/**
+ * Represents a single string on the bass.
+ *
+ * We chose to start at one since musicians count
```

Comment:
Should clarify if 1 is the low string or the high string (closest to the machine or furthest away). It's assumed/conventional that 1 is low but it's not explicit